tests/scripts/all.sh

```diff
@@ -557,6 +563,17 @@ cleanup
 cmake -D CMAKE_BUILD_TYPE:String=Check .
 if_build_succeeded tests/scripts/key-exchanges.pl
 
+if [ $TEST_CXX -ne 0 ]; then
+    msg "build: Unix make, gcc and g++ test" # ~ 30s
+    cleanup
+    make TEST_CPP=1
+
+    msg "build: cmake, gcc and g++ test" # ~ 30s
+    cleanup
+    CC=gcc cmake -D TEST_CPP=YES .
```

Comment:
Why test with both make and cmake? We run some tests with each to validate our build system. But here we're just building a program which has no purpose other than validating that it can be built. Any build system would do.

Reply:
I opted for a larger coverage of build systems, just in case someone forces the CXX compiler for cmake to something not working.

Reply:
Update - Removed the cmake test.